src/java/org/apache/cassandra/db/commitlog/CommitLogSegmentManagerCDC.java

```diff
@@ -250,9 +302,11 @@ void processDiscardedSegment(CommitLogSegment segment)
             {
                 // Add to flushed size before decrementing unflushed so we don't have a window of false generosity
                 if (segment.getCDCState() == CDCState.CONTAINS)
-                    size += segment.onDiskSize();
+                    sizeInProgress.getAndAdd(segment.onDiskSize());
+
+                // Subtract the (estimated) size of the segment from processNewSegment.
                 if (segment.getCDCState() != CDCState.FORBIDDEN)
-                    size -= defaultSegmentSize();
+                    sizeInProgress.getAndAdd(-defaultSegmentSize()); // todo: the defaultSegmentSize can be altered
```

Comment:
It looks like this alteration only takes place in 4 unit tests at this point. Maybe we should take this opportunity to document the `DatabaseDescriptor.setCommitLogSegmentSize` method as being for testing only and point to its usage in `CommitLogSegmentManagerCDC` via javadoc link for future maintainers that may be thinking about changing that?

Reply:
Also, why is this `defaultSegmentSize` instead of just querying the `segment.onDiskSize()` like the call above? And yes, I know the simple answer is "that's because that's how you originally wrote it Josh", but I don't think this is... consistent?

Reply:
The reason to subtract `defaultSegmentSize` is to return the estimated segment size it has aggressively counted at `processNewSegment`. 
In `processDiscardedSegment`, it counts the actual `onDiskSize` for the segment that `CONTAINS`. So it returns the estimated size. For the segment that remains in `PERMITTED`, it gets deleted, so it also need to subtract/return.
I will update the comment to make it more clear.